test/regExpTest.coffee

```diff
@@ -0,0 +1,71 @@
+expect = require('chai').expect
+
+describe 'regExp matches URI', ->
+  regExp = new RegExp("^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?", "i")
```

Comment:
Your results don't make sense though, because I am using a negative lookahead to stop at a space. There isn't a space in the URI presented in your sample, so yeah doesn't make sense. Should look into the flags you are using in the regex